### PR TITLE
feat: add github action to check exact package versions

### PR DESCRIPTION
**Pull request by kyranjamie**
Security recommendation is to ensure packages are locked to a given version. Enforcing no `^`s,  `~`s, `>`s or `<`s in the package.json ensures a stray `yarn` doesn't update the lock file and release an unaudited version.

### CODE REVIEW

File: .github/actions/check-version-lock/check-version-lock.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+const core = require('@actions/core');
```
Comment (zone117x) [inline .github/actions/check-version-lock/check-version-lock.js:1]: This is a neat github action, you should publish it!

File: .github/actions/check-version-lock/check-version-lock.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  try {
+    const dependencies = appPackage.dependencies;
+    const devDependencies = appPackage.devDependencies;
+
+    const allPackages = [...Object.entries(dependencies), ...Object.entries(devDependencies)];
+
+    const caratPackages = allPackages.filter(([packageName, version]) => version.includes('^'));
+    const tildePackages = allPackages.filter(([packageName, version]) => version.includes('~'));
+
+    const illegalPackages = [...caratPackages, ...tildePackages].map(([package, version]) => ({
```
Comment (zone117x) [inline .github/actions/check-version-lock/check-version-lock.js:14]: What about the `>` and `<` comparators?

Comment (kyranjamie) [reply .github/actions/check-version-lock/check-version-lock.js:14]: good point

Comment (kyranjamie) [reply .github/actions/check-version-lock/check-version-lock.js]: done

**Pull request merged**